docs/examples/expdefs/signalsPong.m

```diff
@@ -0,0 +1,338 @@
+
+% ball velocity/angle direction references:
+% at 0 degrees all velocity is in positive X, at 90 degrees all velocity is
+% in positive Y, at 180 degrees all velocity is in negative X, at 270 
+% degrees all velocity is in negative Y
+ballVelX = ballVel * -cos(deg2rad(360-ballAngle));
+ballVelY = ballVel * sin(deg2rad(ballAngle));
+
+% get ball position and exp time at contact
+% use 'merge' to initialize these signals at experiment start
+ballXAtContactOrScore = merge(anyScored.then(0),...
+  ballX.at(contact));
+ballYAtContactOrScore = merge(anyScored.then(0),...
+  ballY.at(contact));
+expTimeAtContactOrScore = merge(anyScored.then(curExpTime-updateTime),...
+  curExpTime.at(contact));
+
+% get ball contact offset values
+ballXContactOffset = cond(ballXAtContactOrScore < 0, ballXAtContactOrScore+1,... 
+  ballXAtContactOrScore > 0, ballXAtContactOrScore-1,... 
+  ballXAtContactOrScore == 0, 0, true, 0);
+ballYContactOffset = cond(ballYAtContactOrScore < 0, ballYAtContactOrScore+1,... 
+  ballYAtContactOrScore > 0, ballYAtContactOrScore-1,... 
+  ballYAtContactOrScore == 0, 0, true, 0);
+
+% get running time from last contact point
+timeDelta = (curExpTime - expTimeAtContactOrScore);
+
+% calculate new velocity and contact offset values after contact
+ballVelXAfterContact = ballVelX.at(timeDelta);
+ballXContactOffsetAfterContact = ballXContactOffset.at(ballVelXAfterContact);
+ballVelYAfterContact = ballVelY.at(timeDelta);
+ballYContactOffsetAfterContact = ballYContactOffset.at(ballVelYAfterContact);
+
+% define mutually dependent signals' interactions
+% ensure the order of updates relating to ball position at a contact point
+% is:
+% 1) 'expTimeAtContact', 2)'ballVel_', 3) 'ball_AtContact'
+ballXToPost = merge( events.newTrial.then(ballInitX),...
+  (timeDelta * ballVelXAfterContact + ballXContactOffsetAfterContact + ... 
+  ballInitX) );
+ballYToPost = merge( events.newTrial.then(ballInitY),...
+  (timeDelta * ballVelYAfterContact + ballYContactOffsetAfterContact + ... 
+  ballInitY) );
+cpuPaddleYToPost = ballYToPost * cpuPaddleCoverage;
+
+% t.Node.Listeners = [t.Node.Listeners,...
+%   ballAngleToPost.delay(updateTime).into(ballAngle),...
+%   ballXToPost.delay(updateTime).into(ballX),... 
+%   ballYToPost.delay(updateTime).into(ballY),...
+%   cpuPaddleYToPost.delay(updateTime).into(cpuPaddleY),...
+%   events.endTrial.onValue(@(~)... 
+%     fprintf('<strong> Score! Player 1: %d cpu: %d </strong>\n',... 
+%     playerScore.Node.CurrValue, cpuScore.Node.CurrValue)),...
+%   events.expStop.onValue(@(~)... 
+%     fprintf('<strong> Game Over! Final Score: Player 1: %d cpu: %d </strong>\n',... 
+%     playerScore.Node.CurrValue, cpuScore.Node.CurrValue)) 
+%     ];  
+
+% use the 't' signal's listeners to update the origin signals 
+% 'ballX', 'ballY', and 'cpuPaddleY'
+% use 'delay' to ensure infinite recursion doesn't occur
+% add listeners each time there is a score, and for when game ends
+t.Node.Listeners = [t.Node.Listeners,...
```

Comment:
For example functions we should always about this hack.  It should be possible to achieve with scan.

Reply:
ok sure, i will work on this then